saml2/saml2-service-provider/src/opensaml3Main/java/org/springframework/security/saml2/provider/service/authentication/OpenSamlAuthenticationProvider.java

```diff
@@ -554,12 +554,19 @@ else if (logger.isDebugEnabled()) {
 			Response response = responseToken.getResponse();
 			Saml2AuthenticationToken token = responseToken.getToken();
 			Saml2ResponseValidatorResult result = Saml2ResponseValidatorResult.success();
+			AbstractSaml2AuthenticationRequest request = token.getAuthenticationRequest();
```

Comment:
Since OpenSAML 3 is EOL, I'd prefer to only add this support to `OpenSaml4AuthenticationProvider`.